### PR TITLE
feat: JumpStartModel attach

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*
Add `JumpStartModel.attach` functionality, to create a `JumpStartModel` from an endpoint name

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by Captainia — APPROVED** [review]

File: src/sagemaker/jumpstart/model.py [hunk 1]
```diff
@@ -402,6 +403,42 @@ def retrieve_example_payload(self) -> JumpStartSerializablePayload:
...
+        The model id, version (and inference component name) can be inferred from the tags.
+        """
+
+        inferred_model_id, inferred_model_version, inferred_inference_component_name = (
+            get_model_id_version_from_endpoint(
+                endpoint_name=endpoint_name,
+                inference_component_name=inference_component_name,
+                sagemaker_session=sagemaker_session,
+            )
+        )
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/model.py:426]: non-blocking: how about skipping this call if a/ `model_id`, `model_version` and `inference_component_name` are specified or b/ `model_id` and `model_version` are specified and the EP is model-based (not IC-based)?

File: tests/unit/sagemaker/jumpstart/model/test_model.py [hunk 2]
```diff
@@ -1292,6 +1292,28 @@ def test_model_artifact_variant_model(
...
+            endpoint_name="some-endpoint",
+            inference_component_name=None,
+            sagemaker_session=DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+        )
+        mock_js_model_init.assert_called_once_with(
+            model_id="model-id",
+            model_version="model-version",
+            sagemaker_session=DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+        )
+        assert isinstance(val, JumpStartModel)
```
Comment (JGuinegagne) [inline tests/unit/sagemaker/jumpstart/model/test_model.py:1315]: could you UT the cases a/ where the `model_id` only is passed, b/ the `model_id`, `model_version` are passed and c/ all of `model_id`, `model_version` and `inference_component_name` are passed by the caller?

File: src/sagemaker/jumpstart/model.py (hunk 1 above)
Comment (evakravi) [reply src/sagemaker/jumpstart/model.py:426]: Good idea. This increases latency unnecessarily

Comment (evakravi) [reply src/sagemaker/jumpstart/model.py:426]: You don't know a priori if the endpoint is IC based, and to figure that out would require an API call, so I will skip this check if all 3 fields are provided

**Review by benieric — APPROVED** [review]

**Pull request merged**